Python/bytecodes.c

```diff
@@ -4349,31 +4353,37 @@ dummy_func(
             res = PyStackRef_FromPyObjectSteal(res_o);
         }
 
-        inst(CALL_ISINSTANCE, (unused/1, unused/2, callable, self_or_null, args[oparg] -- res)) {
-            /* isinstance(o, o2) */
+        op(_GUARD_CALLABLE_ISINSTANCE, (callable, unused, unused, unused -- callable, unused, unused, unused)) {
             PyObject *callable_o = PyStackRef_AsPyObjectBorrow(callable);
-
-            int total_args = oparg;
-            _PyStackRef *arguments = args;
-            if (!PyStackRef_IsNull(self_or_null)) {
-                arguments--;
-                total_args++;
-            }
-            DEOPT_IF(total_args != 2);
             PyInterpreterState *interp = tstate->interp;
             DEOPT_IF(callable_o != interp->callable_cache.isinstance);
+        }
+
+        op(_CALL_ISINSTANCE, (callable, null, inst_, cls -- res)) {
```

Comment:
I don't love the trailing underscore. Not a huge deal, but maybe just rename to `instance` or `obj` or something ?

Reply:
I renamed it to `instance` in [be50e24](https://github.com/python/cpython/pull/133339/commits/be50e24e95b04eb7953c17ef6db21584b2c06a73)